sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py

```diff
@@ -566,6 +566,8 @@ def _download_blob_options(self, offset=None, length=None, validate_content=Fals
             raise ValueError("Encryption required but no key was provided.")
         if length is not None and offset is None:
             raise ValueError("Offset value must not be None if length is set.")
+        if length is not None:
+            length = offset + length - 1  # Service actually uses an end-range inclusive index
```

Comment:
Let's make the comment TODO to rename the length in StorageStreamDownloader to offset_end or range_end?